app/src/androidTest/java/com/sdpteam/connectout/MapViewModelTest.java

```diff
@@ -0,0 +1,105 @@
+package com.sdpteam.connectout;
+
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.MatcherAssert.assertThat;
+
+import android.os.Handler;
+import android.os.Looper;
+
+import androidx.lifecycle.LiveData;
+import androidx.lifecycle.MutableLiveData;
+import androidx.lifecycle.Observer;
+
+import org.hamcrest.CoreMatchers;
+import org.junit.Test;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.TimeUnit;
+
+public class MapViewModelTest {
+    /*@Rule
+    public TestRule rule = new InstantTaskExecutorRule();
+*/
+    @Test
+    public void testMapViewModel() {
```

Comment:
I find that the tests that use LiveData (in `MapViewModelTest`) are hard to rare difficult to read and involve a lot of complex workarounds such as using CountDownLatch and Handler. I propose to find a way to encapsulate all of that into some Utility class.
In fact converting LiveData into a Future could be an ideal solution, as it would simplify tests and make them much easier to read.